cadence/tests/flow_evm_bridge_tests.cdc

```diff
@@ -0,0 +1,501 @@
+    Test.assertEqual(true, aliceIsOwner)
+}
+
+/* --- Script Helpers --- */
+
+access(all)
+fun getCOAAddressHex(atFlowAddress: Address): String {
+    let coaAddressResult = executeScript(
+        "../scripts/evm/get_evm_address_string.cdc",
+        [atFlowAddress]
+    )
+    Test.expect(coaAddressResult, Test.beSucceeded())
+    return coaAddressResult.returnValue as! String? ?? panic("Problem getting COA address as String")
+}
+
+access(all)
+fun getAssociatedEVMAddressHex(with typeIdentifier: String): String {
+    var associatedEVMAddressResult = executeScript(
+        "../scripts/bridge/get_associated_evm_address.cdc",
+        [typeIdentifier]
+    )
+    Test.expect(associatedEVMAddressResult, Test.beSucceeded())
+    return associatedEVMAddressResult.returnValue as! String? ?? panic("Problem getting EVM Address as String")
+}
+
+access(all)
+fun getIDs(ownerAddr: Address, storagePathIdentifier: String): [UInt64] {
+    let idResult = executeScript(
+        "../scripts/nft/get_ids.cdc",
+        [ownerAddr, storagePathIdentifier]
+    )
+    Test.expect(idResult, Test.beSucceeded())
+    return idResult.returnValue as! [UInt64]? ?? panic("Problem getting NFT IDs")
+}
+
+access(all)
+fun isOwner(of: UInt256, ownerEVMAddrHex: String, erc721AddressHex: String): Bool {
+    let isOwnerResult = executeScript(
+        "../scripts/utils/is_owner.cdc",
+        [of, ownerEVMAddrHex, erc721AddressHex]
+    )
+    Test.expect(isOwnerResult, Test.beSucceeded())
+    return isOwnerResult.returnValue as! Bool? ?? panic("Problem getting owner status")
+}
+
+access(all)
+fun deriveBridgedNFTContractName(evmAddressHex: String): String {
+    let nameResult = executeScript(
+        "../scripts/utils/derive_bridged_nft_contract_name.cdc",
+        [evmAddressHex]
+    )
+    Test.expect(nameResult, Test.beSucceeded())
+    return nameResult.returnValue as! String? ?? panic("Problem getting derived contract name")
+}
+
+/* --- Transaction Helpers --- */
+
+access(all)
+fun transferFlow(signer: Test.TestAccount, recipient: Address, amount: UFix64) {
+    let transferResult = executeTransaction(
+        "../transactions/flow-token/transfer_flow.cdc",
+        [recipient, amount],
+        signer
+    )
+    Test.expect(transferResult, Test.beSucceeded())
+}
+
+access(all)
+fun createCOA(signer: Test.TestAccount, fundingAmount: UFix64) {
+    let createCOAResult = executeTransaction(
+        "../transactions/evm/create_account.cdc",
+        [fundingAmount],
+        signer
+    )
+    Test.expect(createCOAResult, Test.beSucceeded())
+}
+
+access(all)
+fun bridgeNFTToEVM(signer: Test.TestAccount, contractAddr: Address, contractName: String, nftID: UInt64) {
+    let bridgeResult = executeTransaction(
+        "../transactions/bridge/bridge_nft_to_evm.cdc",
+        [contractAddr, contractName, nftID],
+        signer
+    )
+    Test.expect(bridgeResult, Test.beSucceeded())
```

Comment:
Can we add a few more lines after this one and for the next test to verify this worked? Could be as simple as using the get ids to make sure the id still exists here in the account on the evm side, and similar for the next test. Also an event emitted check would be good.

Reply:
These are actually helper functions, used in test cases. The actual tests do contain assertions for checking the IDs etc.
I have added some assertions about checking emitted events: https://github.com/onflow/flow-evm-bridge/pull/24/commits/f45ad59ffb028d08e47e9c7eb9bc587d41598c2f